go/framework/bertybridge/bridge.go

```diff
@@ -3,159 +3,341 @@ package bertybridge
 			}
 		}
+		b.clientBridgeService = &client{ccBridge}
+
+		// start Bridge
+		b.logger.Debug("starting Bridge")
+		go func() {
+			b.cerr <- b.workers.Run()
+		}()
+
+		return b, nil
 	}
 
-	// setup lazy grpc listener
-	bl := grpcutil.NewBufListener(ctx, ClientBufferSize)
-	b.workers.Add(func() error {
-		return b.grpcServer.Serve(bl)
-	}, func(error) {
-		bl.Close()
-	})
+	b := Bridge{
+		initManager:        manager,
+		initManagerCleanup: managerCleanup,
+		lcmanager:          manager.GetLifecycleManager(),
+		logger:             logger,
+		notification:       notifmanager,
+	}
 
-	// setup lazy grpc listener for services
-	var ccServices *grpc.ClientConn
-	var bufListener *grpcutil.BufListener
-	{
-		var err error
+	return &b, nil
+}
 
-		bufListener = grpcutil.NewBufListener(ctx, ClientBufferSize)
-		b.workers.Add(func() error {
-			return b.grpcServer.Serve(bufListener)
-		}, func(error) {
-			bufListener.Close()
-		})
+func (b *Bridge) HandleState(appstate int) {
+	b.muAppState.Lock()
+	defer b.muAppState.Unlock()
+
+	if appstate != b.currentAppState {
+		switch appstate {
+		case AppStateBackground:
+			b.lcmanager.UpdateState(bertymessenger.StateInactive)
+			b.logger.Info("app is in Background State")
+		case AppStateActive:
+			b.lcmanager.UpdateState(bertymessenger.StateActive)
+			b.logger.Info("app is in Active State")
+			if b.node != nil {
+				bootstrapConfig := bootstrap.BootstrapConfig{
+					MinPeerThreshold:  5,
+					Period:            10 * time.Second,
+					ConnectionTimeout: 5 * time.Second, // Period / 2
+				}
+				if err := b.node.Bootstrap(bootstrapConfig); err != nil {
+					b.logger.Warn("Unable to boostrap node", zap.Error(err))
+				}
+			}
+		case AppStateInactive:
+			b.lcmanager.UpdateState(bertymessenger.StateInactive)
+			b.logger.Info("app is in Inactive State")
+		}
+		b.currentAppState = appstate
+	}
+}
+
+var backgroundCounter int32
+
+func (b *Bridge) HandleTask() LifeCycleBackgroundTask {
+	return newBackgroundTask(b.logger, func(ctx context.Context) error {
+		b.logger.Info("starting background task")
+
+		counter := atomic.AddInt32(&backgroundCounter, 1)
+		tnow := time.Now()
+
+		n := time.Duration(mrand.Intn(60) + 5) // nolint:gosec
+		ctx, cancel := context.WithTimeout(ctx, time.Second*n)
+		defer cancel()
+
+		if err := b.notification.Notify(&notification.Notification{
+			Title: fmt.Sprintf("GoBackgroundTask #%d", counter),
+			Body:  "started",
+		}); err != nil {
+			b.logger.Error("unable to notify", zap.Error(err))
+		}
 
-		if ccServices, err = bufListener.NewClientConn(); err != nil {
-			return nil, errors.Wrap(err, "unable to get services gRPC ClientConn")
+		<-ctx.Done()
+
+		if err := b.notification.Notify(&notification.Notification{
+			Title: fmt.Sprintf("GoBackgroundTask #%d", counter),
+			Body:  fmt.Sprintf("ended (duration: %s)", time.Since(tnow).Truncate(time.Second)),
+		}); err != nil {
+			b.logger.Error("unable to notify", zap.Error(err))
```

Comment:
can you remove/comment `b.notification.Notify` along the counter? It should not be usefull anymore